internal/painter/software/draw.go

```diff
@@ -19,16 +21,22 @@ type gradient interface {
 	Size() fyne.Size
 }
 
-func drawCircle(c fyne.Canvas, circle *canvas.Circle, pos fyne.Position, base *image.NRGBA, clip image.Rectangle) {
+func (p *Painter) drawCircle(c fyne.Canvas, circle *canvas.Circle, pos fyne.Position, base *image.NRGBA, clip image.Rectangle) {
+	t := time.Now()
 	pad := painter.VectorPad(circle)
 	scaledWidth := scale.ToScreenCoordinate(c, circle.Size().Width+pad*2)
 	scaledHeight := scale.ToScreenCoordinate(c, circle.Size().Height+pad*2)
 	scaledX, scaledY := scale.ToScreenCoordinate(c, pos.X-pad), scale.ToScreenCoordinate(c, pos.Y-pad)
 	bounds := clip.Intersect(image.Rect(scaledX, scaledY, scaledX+scaledWidth, scaledY+scaledHeight))
 
-	raw := painter.DrawCircle(circle, pad, func(in float32) float32 {
-		return float32(math.Round(float64(in) * float64(c.Scale())))
+	raw, err := p.getTexture(circle, func(object fyne.CanvasObject) Texture {
```

Comment:
This is a problem for tests (wrong type for `TextureType`), and I'm not quite sure what's a good way around it...

Maybe we could make the painter functions internal and add a wrapper file (i.e. `draw_extern_software.go`, `draw_extern.go`) that handles storing in the cache (or does nothing)? being that it's only a few lines, it should be eligible for inlining, so the wrapper should be functionally free.